tests/test_data_utils.py

```diff
@@ -0,0 +1,169 @@
+
+
+@pytest.fixture(scope="module")
+def dummy_1040():
+    return pd.DataFrame(
+        {
+            "simulant_id": list(range(8)) * 2,
+            "joint_filer": [
+                False,
+                True,
+                False,
+                False,
+                True,
+                False,
+                False,
+                False,
+            ]
+            * 2,
+            "relationship_to_reference_person": [
+                "Reference person",
+                "Opp-sex spouse",
+                "Biological child",
+                "Reference person",
+                "Opp-sex spouse",
+                "Reference person",
+                "Roommate",
+                "Opp-sex spouse",
+            ]
+            * 2,
+            "household_id": [
+                10,
+                10,
+                10,
+                11,
+                11,
+                12,
+                13,
+                14,
+            ]
+            * 2,
+            "tax_year": [2020] * 8 + [2021] * 8,
+            "ssn_itin": ["123-45-6789"] * 16,
+            "copy_ssn": ["000-00-0000"] * 16,
+        }
+    )
+
+
+@pytest.fixture(scope="module")
+def dummy_tax_dependents():
+    return pd.DataFrame(
+        {
+            "simulant_id": [2, 103, 104, 105, 106, 107, 108, 109] * 2,
+            "guardian_id": [0, 0, 0, 0, 0, 3, 3, 5] * 2,
+            "favorite_food": [
+                "Pizza",
+                "Cookie",
+                "Ice cream",
+                "Cheeseburger",
+                "Sandwich",
+                "Salad",
+                "Tacos",
+                "Pasta",
+                "Ramen",
+                "Waffles",
+                "Cookies",
+                "Watermelon",
+                "Nachos",
+                "BBQ",
+                "Bagel",
+                "Grapes",
+            ],
+            "tax_year": [2020] * 8 + [2021] * 8,
+            "first_name": ["Fake First name"] * 16,
+            "last_name": ["Secret last name"] * 16,
+            "ssn_itin": ["987-65-4321"] * 16,
+            "copy_ssn": ["111-22-3333"] * 16,
+        }
+    )
+
+
+def test_combine_joint_filers(dummy_1040):
+    joint_1040 = combine_joint_filers(dummy_1040)
+
+    assert set(joint_1040.columns) == set(
+        [
+            "simulant_id",
+            "relationship_to_reference_person",
+            "joint_filer",
+            "household_id",
+            "tax_year",
+            "ssn_itin",
+            "copy_ssn",
+            "spouse_simulant_id",
+            "spouse_relationship_to_reference_person",
+            "spouse_joint_filer",
+            "spouse_tax_year",
+            "spouse_household_id",
+            "spouse_ssn_itin",
+            "spouse_copy_ssn",
+        ]
+    )
+    joint_filer_ids = dummy_1040.loc[dummy_1040["joint_filer"] == True, "simulant_id"]
+    # Joint filer ids should not be in simulant ids
+    assert not bool(set(joint_1040["simulant_id"]) & set(joint_filer_ids))
```

Comment:
You don't actually need to cast the thing as a bool.

Reply:
e.g. 

>>> assert not {1,2,3} & {4,5,6}
>>> assert not {1,2,3} & {4,5,6,1}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>>